### ISSUE TITLE
Rule request: Enforce only tabs OR spaces

### ISSUE THREAD

**Issue by d12bb** [realm/SwiftLint#3839]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request
As this is in my eyes a pretty obvious thing, please just close if there's an existing rule for this. I searched extensively and didn't find a rule or existing issue.

As far as I can tell, the indentation_width rule only enforces use of consistent amount of spaces or tabs. However, depending on an editor's set tab width, this can lead to inconsistent indentation when one developer uses tabs, another spaces. So swiftlint should enforce that only one style of indentation is used (and preferably enforce tabs, of course 😬).

Should trigger warning (as GitHub uses a tab width of eight, it's even easy to see here):
```swift
private func test() {
	print("One Tab")
    print("Four Spaces")
}
```

Current output:
```shell
$ swiftlint --enable-all-rules test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

I think this should be opt-out, by default it should when mixing tabs and spaces and one should be able to configure what shall be used something like this:
```yaml
indentation_style: tabs   -> warn when indenting with spaces
indentation_style: spaces -> warn when indenting with tabs
```
